src/MyTree.tsx

```diff
@@ -1,42 +1,22 @@
+import React from "react";
 import { Tree } from "antd";
+
 import { Schema } from "./data";
-import {
-  FormOutlined,
-  TableOutlined,
-  FilterOutlined,
-  NumberOutlined,
-} from "@ant-design/icons";
-import React from "react";
+import { search } from "./search";
+import { dataFormatter, treeFormatter } from "./formatter";
 
 type MyTreeProps = {
   data: Schema[];
+  searchTerm?: string;
+  onlyPublic?: boolean;
+  showIcons?: boolean;
+  withSystemName?: boolean;
 };
 
-const FormIcon = FormOutlined;
-const TableIcon = TableOutlined;
-const ViewIcon = FilterOutlined;
-const ColumnIcon = NumberOutlined;
-
-// If you don't want to use typescript, use line 10 instead of 11 and comment out the type `MyTreeProps` above
-// export const MyTree = (props) => {
 export const MyTree = (props: MyTreeProps) => {
-  const sampleData = [
-    {
-      title: "Sample Data",
-      key: "sample",
-      icon: <FormIcon />,
-      children: [
-        { title: "Sample Child 1", key: "sample child 1", icon: <ViewIcon /> },
-        {
-          title: "Sample Child 2",
-          key: "sample child 2",
-        },
-      ],
-    },
-  ];
-
-  // TODO - replace sampleData with testData transformed from props
-  const treeData = sampleData;
+  const formattedData = dataFormatter(props.data);
+  const filteredData = search(formattedData, props.searchTerm, props.onlyPublic);
+  const treeData = treeFormatter(filteredData, props.showIcons, props.withSystemName);
```

Comment:
If I were to redo this exercise, I think I would make a Formatter class which would take the data when initialized, and then have functions like "formatAsTree". Then this code would live together where it belongs.